src/main/java/org/jabref/logic/linkedfile/LinkedFileNamePattern.java

```diff
@@ -0,0 +1,105 @@
+package org.jabref.logic.linkedfile;
+
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public record LinkedFileNamePattern(String stringRepresentation, Category category) {
+    public enum Category {
+        AUTHOR_RELATED, EDITOR_RELATED, TITLE_RELATED, OTHER_FIELDS, BIBENTRY_FIELDS
+    }
+
+    public static final LinkedFileNamePattern NULL_LINKED_FILE_NAME_PATTERN = new LinkedFileNamePattern("", Category.OTHER_FIELDS);
+
+    // region - Author-related field markers
+    public static final LinkedFileNamePattern AUTHOR_YEAR = new LinkedFileNamePattern("[auth_year]", Category.AUTHOR_RELATED);
```

Comment:
This is basically a duplicate of the CitationKeyPatterns. Why not simply reuse them?
Or mabye just a simple wrapper class around them